.github/workflows/ci.yml

```diff
@@ -12,22 +12,15 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - name: Set up Go 1.20
-        uses: actions/setup-go@v4
-        with:
-          go-version: '1.20'
-        id: go
-
       - name: Check out code into the Go module directory
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
-      - name: Restore cache
```

Comment:
Why did you delete cache action ?

Reply:
It's redundant because `setup-go@v4` has enabled caching by default https://github.com/actions/setup-go#v4

Reply:
let's proceed in this direction, once. In fact, it's better no to use setup-go cache and use action/cache/restore and action/cache/save, but this problem is absolutely another thing. So let's handle it separately. Thanks.